lib/httpclient/http.rb

```diff
@@ -574,17 +574,19 @@ def reset_pos(io)
       end
 
       def dump_file(io, dev, sz)
+
         buf = ''
         rest = sz
         while rest > 0
           n = io.read([rest, @chunk_size].min, buf)
-          raise ArgumentError.new("Illegal size value: #size returns #{sz} but cannot read") if n.nil?
```

Comment:
Yes, it's intentional to detect the situation where "it should be able to read XX bytes but httpclient can read only <XX bytes". I didn't want to ignore this unexpected situation.
How does sz become unexpected number in your case?